Add initial ADR records

neilbmclaughlin opened the pull request.

neilbmclaughlin committed:
:movie_camera: Add initial ADR record

c2s-dev commented:
:rocket: deployment of profiles-db-elastic succeeded (http://profiles-db-elastic-pr-32.dev.beta.nhschoices.net)

c2s-dev commented:
:rocket: deployment of profiles-db-elastic succeeded (http://profiles-db-elastic-pr-32.dev.beta.nhschoices.net)

c2s-dev commented:
:rocket: deployment of profiles-db-elastic succeeded (http://profiles-db-elastic-pr-32.dev.beta.nhschoices.net)

c2s-dev commented:
:rocket: deployment of profiles-db-elastic succeeded (http://profiles-db-elastic-pr-32.dev.beta.nhschoices.net)

c2s-dev commented:
:rocket: deployment of profiles-db-elastic succeeded (http://profiles-db-elastic-pr-32.dev.beta.nhschoices.net)

neilbmclaughlin committed:
:movie_camera: Add ADR for ES Docker Image

neilbmclaughlin committed:
:movie_camera: Add ADR for GP data load.

c2s-dev commented:
:rocket: deployment of profiles-db-elastic succeeded (http://profiles-db-elastic-pr-32.dev.beta.nhschoices.net)

st3v3nhunt commented on doc/architecture/decisions/0002-use-elastic-co-docker-image.md:
The latter part of this sentence seems more like a consequence to me. Perhaps rewrite to `We will use the official Docker image provided by elastic.co`. I don't know if the bit about initial POCs and early Beta is required, especially as these things tend to no just be for that initial period.

st3v3nhunt commented on doc/architecture/decisions/0002-use-elastic-co-docker-image.md:
I think the point above about having to use X-Pack to change the password is the thing that needs recording. I think this line i.e. line 23 can be removed.

st3v3nhunt commented on doc/architecture/decisions/0002-use-elastic-co-docker-image.md:
Suggest changing `going forward` to `before the service is presented to users by default` (or some other time we deem it needs to be addressed by).

st3v3nhunt commented on doc/architecture/decisions/0002-use-elastic-co-docker-image.md:
Perhaps some rewording here? It sounds like the consequence of using the elastic.co image is that there are additional challenges in managing different versions of ES compared to using the Docker hub version.

Is that actually the case? Certainly the knock on effect of having images with X-Pack baked in is a consequence that needs recording (which is done below).

st3v3nhunt commented on doc/architecture/decisions/0003-bulk-load-gp-data.md:
```diff
@@ -0,0 +1,23 @@
+
+## Context
+
+In the early stages of ES adoption (which predates the development of ETL processes to automatically build the GP JSON data file) the GP data mas manually added to this repo. The data is in a format optimised for loading into MongoDB. Whilst Elastic Search (ES) can import documents singly in the format it takes some considerable time. By re-shaping the data to that required by the bulk import API ...
```
I think the context can be simplified quite a bit to something like `Other work within the same domain has created a JSON file containing all GP practices that need to be searched and exposed via ES. This datafile was originally created to be loaded into MongoDB.` (or something equally or more succinct.

st3v3nhunt commented on doc/architecture/decisions/0003-bulk-load-gp-data.md:
```diff
@@ -0,0 +1,23 @@
+
+## Decision
+
+The use of an existing data file coupled with pre-procesing (using a `jq` script) to reshape it and subsequent bulk loading it into ES was a pragmatic move
```
Suggest rewording to `We will use the existing data file but reshape the data specifically to suit the needs of ES via jq (a lightweight JSON processing cli tool).` (or something similar).

st3v3nhunt commented on doc/architecture/decisions/0003-bulk-load-gp-data.md:
```diff
@@ -0,0 +1,23 @@
+ 
+## Consequences
+
+The data is loaded using bash scripts run within the container as part of `docker build` which means that the resulting Docker image contains ES preloaded with the GP data.
```
I don't think this needs to be here.

st3v3nhunt commented on doc/architecture/decisions/0003-bulk-load-gp-data.md:
```diff
@@ -0,0 +1,23 @@
+
+The data is loaded using bash scripts run within the container as part of `docker build` which means that the resulting Docker image contains ES preloaded with the GP data.
+
+The updating of the source GP data file in this repo is a manual process with no set schedule. This will likely be superceeded by the GP data ETL processors in the near future.
```
👍

st3v3nhunt commented on doc/architecture/decisions/0003-bulk-load-gp-data.md:
```diff
@@ -0,0 +1,23 @@
+
+The updating of the source GP data file in this repo is a manual process with no set schedule. This will likely be superceeded by the GP data ETL processors in the near future.
+
+As well as reshaping the GP JSON document for bulk insert some manipulation of the properties was necessary to optimise for particular search requirements. The reshaping was carried out using `jq` as a lightweight way of doing this but did mean that there are no tests around the reshaping.
```
👍

st3v3nhunt reviewed (changes requested):
A few comments

neilbmclaughlin committed:
:bowtie: Rewording of ADRs for clarity

c2s-dev commented:
:rocket: deployment of profiles-db-elastic succeeded (http://profiles-db-elastic-pr-32.dev.beta.nhschoices.net)

c2s-dev commented:
:rocket: deployment of profiles-db-elastic succeeded (http://profiles-db-elastic-pr-32.dev.beta.nhschoices.net)

c2s-dev commented:
:rocket: deployment of profiles-db-elastic succeeded (http://profiles-db-elastic-pr-32.dev.beta.nhschoices.net)

c2s-dev commented:
:rocket: deployment of profiles-db-elastic succeeded (http://profiles-db-elastic-pr-32.dev.beta.nhschoices.net)

c2s-dev commented:
:rocket: deployment of profiles-db-elastic succeeded (http://profiles-db-elastic-pr-32.dev.beta.nhschoices.net)

neilbmclaughlin commented:
@st3v3nhunt - I've reworked the ADR's in light of your comments and a re-reading of the Michael Nygard post. I've also split the decision record around usage of the ES Docker image into one for Docker usage in general and one for using the ES provided one specifically.

c2s-dev commented:
:rocket: deployment of profiles-db-elastic succeeded (http://profiles-db-elastic-pr-32.dev.beta.nhschoices.net)

c2s-dev commented:
:rocket: deployment of profiles-db-elastic succeeded (http://profiles-db-elastic-pr-32.dev.beta.nhschoices.net)

st3v3nhunt commented on doc/architecture/decisions/0004-bulk-load-gp-data.md:
```diff
@@ -0,0 +1,23 @@
+# 3. Build Image with GP Data
```
3 should be a 4

neilbmclaughlin replied:
@st3v3nhunt fixed and ready for merge.

st3v3nhunt reviewed (changes requested):
Looking good. Just a small typo I noticed.

neilbmclaughlin committed:
:bug: Fix of typo

c2s-dev commented:
:rocket: deployment of profiles-db-elastic succeeded (http://profiles-db-elastic-pr-32.dev.beta.nhschoices.net)
